File: 0125-valid-palindrome/0125-valid-palindrome.py
class Solution(object):
    def isPalindrome(self, s):
        """
        :type s: str
        :rtype: bool
        """
        n = len(s)
        main = ""
        s = s.lower()
        check = "abcdefghijklmnopqrstuvwxyz1234567890"
        for char in s:
            if char in check:
                main+=char
        return main == main[::-1]
        

File: 0125-valid-palindrome/0125-valid-palindrome.cpp
class Solution {
public:
    bool isPalindrome(string s) {
        int len = s.length();
        string newS = "";

        for(int i=0;i<len;i++){
            if(isalnum(s[i])){
                newS+=tolower(s[i]);
            }
        }
        std::cout<<newS;
        int left = 0;
        int right = newS.length()-1;
        while (left<=right){
            if (newS[left]!=newS[right]){
                return false;
            } else {
                left++;
                right--;
            }
        }

        return true;
    }
};